src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2951,11 +2951,7 @@ private TypeCompareState isEnum(CORINFO_CLASS_STRUCT_* cls, CORINFO_CLASS_STRUCT
             }
 
             TypeDesc type = HandleToObject(cls);
-
-            if (type.IsGenericParameter)
-            {
-                return TypeCompareState.May;
-            }
+            Debug.Assert(!type.IsGenericParameter);
```

Comment:
I do not think it is rechargeable today, but it may become reachable if we keep pushing the boundaries for the reflection APIs handled as JIT intrinsics.

Reply:
Is your suggestion to close this and keep this "just in case"? We've had this dead code here for two years and I'm not aware of any other JitInterface API being able to deal with generic parameters.

Reply:
Either way is fine with me. Whatever we decide on, we should do it consistently.

Reply:
> Either way is fine with me. Whatever we decide on, we should do it consistently.

Making it consistent in https://github.com/dotnet/runtime/pull/101108.